gap_windows.go

```diff
@@ -18,6 +20,119 @@ type Address struct {
 	MACAddress
 }
 
+type Advertisement struct {
+	advertisement *advertisement.BluetoothLEAdvertisement
+	publisher     *advertisement.BluetoothLEAdvertisementPublisher
+}
+
+// DefaultAdvertisement returns the default advertisement instance but does not
+// configure it.
+func (a *Adapter) DefaultAdvertisement() *Advertisement {
+	if a.defaultAdvertisement == nil {
+		a.defaultAdvertisement = &Advertisement{}
+	}
+
+	return a.defaultAdvertisement
+}
+
+// Configure this advertisement.
+// on Windows we're only able to set "Manufacturer Data" for advertisements.
+// https://learn.microsoft.com/en-us/uwp/api/windows.devices.bluetooth.advertisement.bluetoothleadvertisementpublisher?view=winrt-22621#remarks
+// following this c# source for this implementation: https://github.com/microsoft/Windows-universal-samples/blob/main/Samples/BluetoothAdvertisement/cs/Scenario2_Publisher.xaml.cs
+// adding service data / localname leads to errors when starting the advertisement.
+func (a *Advertisement) Configure(options AdvertisementOptions) error {
+	if a.advertisement != nil {
+		panic("todo: configure advertisement a second time")
+	}
+
+	// we can only advertise manufacturer / company data on windows, so no need to continue if we have none
+	if len(options.ManufacturerData) == 0 {
+		return nil
+	}
+
+	pub, err := advertisement.NewBluetoothLEAdvertisementPublisher()
+	if err != nil {
+		return err
+	}
+
+	a.publisher = pub
```

Comment:
Good point, added that for the publisher as well as the advertisement, which was unused by accident! Looks to be working in the heartrate example if I try to reconfigure it a couple times before starting.

Could you give some more insight on how the advertisement is switching? The advertisements were quite strange to get working, I copied most of their implementation for this C# [source](https://github.com/microsoft/Windows-universal-samples/blob/main/Samples/BluetoothAdvertisement/cs/Scenario2_Publisher.xaml.cs), which I was never really able to see from my phone using the [ble tester](https://play.google.com/store/apps/details?id=com.zhctwh.ble_tester&pli=1) app on android. But seems to work fine when running the heartrate example.

Reply:
ah okay when using nrf connect I can see it. It's not alternating for me, sometimes it pops up then disappears for a second or two though, but I never see any windows advertising data.

this behavior is the same when I run the C# source as well, seems to be a windows quirk?
when running the heartrate example, and adding some manufacturer data to the advertisement, it shows up steady

Reply:
> Could you give some more insight on how the advertisement is switching?

This is pretty common behavior. If there's more than one advertisement happening at the same time, many systems will alternate between them. Because they use the same MAC address for both advertisements, it looks like it's constantly changing.
I wouldn't worry about it if it doesn't cause any problems.